astroquery/utils/tap/core.py

```diff
@@ -217,28 +291,40 @@ def launch_job(self, query, name=None, output_file=None,
                 response.getheaders(),
                 "location")
             if location is None:
-                raise requests.exceptions.HTTPError("No location found after redirection was received (303)")
+                raise requests.exceptions.HTTPError("No location found "
+                                                    "after redirection was "
+                                                    "received (303)")
             if verbose:
                 print("Redirect to %s", location)
             subcontext = self.__extract_sync_subcontext(location)
-            response = self.__connHandler.execute_get(subcontext)
+            response = self.__connHandler.execute_tapget(subcontext,
+                                                         verbose=verbose)
         job = Job(async_job=False, query=query, connhandler=self.__connHandler)
         isError = self.__connHandler.check_launch_response_status(response,
                                                                   verbose,
-                                                                  200)
+                                                                  200,
+                                                                  False)
+        headers = response.getheaders()
         suitableOutputFile = self.__getSuitableOutputFile(False,

```

Comment:
To add support for tsv format in the tapconn.py file in the function get_suitable_extension_by_format() could you add if tsv return .tsv and the in function get_suitable_extension() could you add if tab-separated-values return .tsv